tests/ignite/base/test_events_driven.py

```diff
@@ -0,0 +1,238 @@
+import pytest
+
+from ignite.base.events_driven import EventsDriven, EventsDrivenState
+from ignite.engine.events import EventEnum
+
+
+class EventsDrivenWithState(EventsDriven):
+    def __init__(self) -> None:
+        super(EventsDrivenWithState, self).__init__()
+        self._state = EventsDrivenState(engine=self)
+
+    @property
+    def state(self) -> EventsDrivenState:
+        return self._state
+
+    def register_events(self, *event_names, event_to_attr=None) -> None:
+        super(EventsDrivenWithState, self).register_events(*event_names)
+        self._state.update_mapping(event_to_attr)
+
+
+class ABCEvents(EventEnum):
+    A_EVENT = "a_event"
+    B_EVENT = "b_event"
+    C_EVENT = "c_event"
+
+
+# ! There are some updates to be done in EventsDriven
+# ! uncomment this when EventsDriven is updated in the upcoming PR
+# def test_events_driven_basics():
```

Comment:
Which updates you are talking about here? I think this test file should contain all tests. If it is not the case, maybe something is wrong

Reply:
Sorry I should have made that more clear in the comments, currently we are fetching the event by the same old way in the (using state) `_handler_wrapper` https://github.com/pytorch/ignite/blob/44a64d6d8def637ae636471faf1487a18d209456/ignite/base/events_driven.py#L49, which is will make this test fail. Why don't we update it and fetch the event using `_allowed_event_counts` as we should do in the upcoming updates? using `_allowed_event_counts` without the other necessary updates along with it will make `test_deterministic` fails, so I commented it for now, and as soon as we update the engine and `_handler_wrapper` I will uncomment this.

Reply:
> Why don't we update it and fetch the event using _allowed_event_counts as we should do in the upcoming updates?

Yes, this is also my question :) Ahmed, it is your PR


I commented before to remove EventsDriven from Engine and in this PR we should just introduce EventsDriven...
All tests should be present and passing.

Reply:
> using _allowed_event_counts without the other necessary updates along with it will make test_deterministic fails, so I commented it for now, and as soon as we update the engine and _handler_wrapper I will uncomment this.

That's the answer for the question. I understand your point now, I will revert the engine updates.